.github/workflows/ci.yaml

```diff
@@ -20,6 +22,135 @@ jobs:
+          done
+          echo "changed-files=$(echo "${values}" | jq -R -s -c 'split("\n")[:-1]')" >> $GITHUB_OUTPUT
+
+      - name: Read Falco versions
+        id: set-falco-versions
+        run: |
+          values=""
+          while read -r line
+          do
+            values="${values}${line}"$'\n'
+          done < "./.github/FALCO_VERSION"
+          echo "versions=$(echo "${values}" | jq -R -s -c 'split("\n")[:-2]')" >> $GITHUB_OUTPUT
+
+  rules-validate:
+    needs: rules-get-values
+    strategy:
+      fail-fast: false
+      matrix:
+        rules-file: ${{ fromJson(needs.rules-get-values.outputs.changed-files) }}
+        falco-version: ${{ fromJson(needs.rules-get-values.outputs.falco-versions) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Golang
+        uses: actions/setup-go@v3
+        with:
+          go-version: '^1.19'
+
+      - name: Checkout rules
+        uses: actions/checkout@v3
+
+      - name: Build checker tool
+        working-directory: build/checker
+        run: go build -o rules-check
+      
+      - name: Test checker tool
+        working-directory: build/checker
+        run: go test ./... -cover
+
+      - name: Validate rules file
+        run: |
+          build/checker/rules-check \
+              validate \
+              --falco-image="falcosecurity/falco-no-driver:${{ matrix.falco-version }}" \
+              ${{ matrix.rules-file }}
+
+  rules-check-version:
+    needs: rules-get-values
+    env:
+      # note(jasondellaluce): using the most recent targeted Falco version
+      FALCO_VERSION: ${{ fromJson(needs.rules-get-values.outputs.falco-versions)[0] }}
+    strategy:
+      fail-fast: false
+      matrix:
+        rules-file: ${{ fromJson(needs.rules-get-values.outputs.changed-files) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Golang
+        uses: actions/setup-go@v3
+        with:
+          go-version: '^1.19'
+
+      - name: Checkout rules
+        uses: actions/checkout@v3
+
+      - name: Get all git tags
+        run: git fetch --tags origin
+
+      - name: Get changed files
+        uses: jitterbit/get-changed-files@v1
+        id: changed
+        with:
+          format: space-delimited
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build checker tool
+        working-directory: build/checker
+        run: go build -o rules-check
+      
+      - name: Test checker tool
+        working-directory: build/checker
+        run: go test ./... -cover
+
+      - name: Compare changed files with previous versions
+        id: compare
+        run: |
+          ./.github/compare-rule-files.sh \
+              ${{ matrix.rules-file }} \
+              result.txt \
+              build/checker/rules-check \
+              "falcosecurity/falco-no-driver:$FALCO_VERSION"
+          if [ -s result.txt ]; then
+            echo "comment_file=result.txt" >> $GITHUB_OUTPUT
+          fi
+
+      - uses: mshick/add-pr-comment@v2
```

Comment:
This is great! However, i am a bit worried if this will work when run on a PR coming from a fork. I had similar issues trying to post a comment on libs repo and had to move to `pull_request_target` trigger that is unsafe: https://github.com/falcosecurity/libs/pull/977

Reply:
That's it: https://github.com/mshick/add-pr-comment#proxy-for-fork-based-prs